### DISCUSSION TITLE
Versioning implementation

### DISCUSSION THREAD

**Discussion by stepin105361** [argoproj/argo-cd#7379] [Q&A]
How did you guys design the version selector? Incorporating it with the .md files in the header of the page? Because if i remember, in the material theme using the readthedocs, it doesnt support that version selector naturally. So you have actually coded the part. I am doing the same thing. As of now i have got multiple versions of my doc live via readthedocs. I am using the material theme. So can you tell me how to implement the version selector.
Some help would be really appreciated.

**Comment by alexmt** [comment]
Hello @stepin105361 ,

We have used `extra_javascript` and `extra_css` mkdocs settings to attach javascript/css files that renders version selector:

https://github.com/argoproj/argo-cd/blob/b0736219b6bd3bcba3b6339da21f0024d17e09dd/mkdocs.yml#L14-L17

The [versions.js](https://github.com/argoproj/argo-cd/blob/master/docs/assets/versions.js)  and [versions.css](https://github.com/argoproj/argo-cd/blob/master/docs/assets/versions.css) implements selector rendering. You might copy these files in your project. Don't forgot to replace `https://argo-cd.readthedocs.io` with your repository URL in versions.js file

**Reply by stepin105361** [reply]
<img width="912" alt="Screenshot 2021-10-08 131620" src="https://user-images.githubusercontent.com/69123968/136518450-75d894aa-a040-4947-817a-ea97f86e8742.png">

Hey @alexmt , after the callbackName + 
you have written some proj-argo-cd and the link, what to put there for my site? becuase i dont see anywhere your repo or site where the name is like this : '&project=argo-cd&page=&theme=mkdocs&format=jsonp&docroot=docs&source_suffix=.md&version='
apologies if my doubt is not going through, do tell and i will explain myself more.

**Comment by stepin105361** [comment]
Hey @alexmt I implemented the version selector but in my case the drop down with selector is not coming up dont know why. URL is mine and also multiple versions are active in the readthedocs.io. So why am i not able to reproduce the same dropdown effect on my page?
